proposals/distribution.md

```diff
@@ -0,0 +1,66 @@
+
+This proposal covers the distribution API spec, and while it does not cover the code for the docker-registry, that implementation is considered the reference implementation. There are other implementations of this protocol, not all are open-source though (Google gcr.io, Amazon ECR, CoreOS Quay, Gitlab registry, JFrog Artifactory registry, Huawei Dockyard, etc).
+
+In the past when the topic of having an OCI specification around the distribution of container images was discussed, it was deferred as "let’s get the image format defined, meanwhile the industry will settle on a distribution standard". Fast forward, OCI image format is out and adopted, and the Registry v2 is the defacto standard. There is and will be use-cases for alternate methods and the future will likely hold creative ways to push, fetch and share container images, but right now this promotion serves to acknowledge by the OCI the current industry standard of distributing container images.
+
+There is polish that is needed e.g. broken links to storage-driver docs, as well as making sections more generic regarding the OCI descriptors and media-types, but on the whole this is a lateral move.
+
+## Initial Maintainers
+
+* Stephen Day <stephen.day@docker.com> (@stevvooe)
+* Vincent Batts <vbatts@redhat.com> (@vbatts)
+* Derek McGowan <derek.mcgowan@docker.com> (@dmcgowan)
+
+Additional Maintainers to consider:
+
+* Ahmet Alp Balkan (Google)
+* Matt Moore (Google)
+* Yuwa (MSFT)
+* Clayton Coleman (Red Hat)
+* Antonio Murdaca (@runcom) (Red Hat)
+* Samuel Karp (@samuelkarp) (AWS)
+* Mike Brown (IBM)
+* Jimmy Zelinskie jimmy@coreos.com (@jzelinskie)
+* Liu Genping <[liugenping@huawei.com](mailto:liugenping@huawei.com)>
+* Vanessa Sochat (@vsoch) (Stanford) <vsochat@stanford.edu>
+* Eduardo Arango (@ArangoGutierrez) (Sylabs) <eduardo@sylabs.io>
+
+## Code of Conduct
+
+This project would incorporate (by reference) the OCI Code of Conduct ([https://github.com/opencontainers/tob/blob/master/code-of-conduct.md](https://github.com/opencontainers/tob/blob/master/code-of-conduct.md)).
+
+## Governance and Releases
+
+This project would incorporate the Governance and Releases processes from the OCI project template: [https://github.com/opencontainers/project-template](https://github.com/opencontainers/project-template).
+
+## Project Communications
+
+Both of the proposed projects would continue to use existing channels in use by the OCI developer community for communication including:
+
+* GitHub for issues and pull requests
+* The dev@opencontainers.org email list
+* The monthly OCI developer community conference call
+* The #OpenContainers freenode IRC channel
+
+## Versioning / Roadmap
+
+The API spec is currently considered v2 and we will start the specification at v2.0. Fewer places to change and compare, and it would keep with it being a lateral move.
```

Comment:
I'm not sure we should start with `v2.0`, as the latest `Docker-Distribution-API-Version` header is set to `registry/2.0` (https://docs.docker.com/registry/spec/api/#introduction)

How about starting with `V2 Release 1.0"?

Reply:
Or just `v2.1`

Reply:
> Or just v2.1

Somewhat complicating this approach is that the fact that Docker's v2 API has been receiving [lettered minor/patch bumps][1].  Presumably those will all be considered part of the 2.0.x series?

And if the goal is to start using SemVer (which sounds like a useful goal to me), we'll need to specify that and wait for some deprecation period before cutting v2.1, or we'll have all the 2.0 clients choke and die when the [version-check endpoint][2] sets `Docker-Distribution-API-Version` to `registry/2.1`.  We want them to all be able to say “I'm 2.0 compliant, so a 2.1 registry is fine”.

[1]: https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md#changes
[2]: https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md#api-version-check

Reply:
@wking In general, since the subversions are backwards compatible, we do not expose the subversions. In general, we should move away from path-level versioning and favor type-level versioning.

Reply:
@akihirosuda are you saying for that header, or just tagged releases or both?

Reply:
FWIW, Docker v2-2 is the most recently protocol -- I hope everyone here knows that and has just been using 2.1 as an example.